sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -218,16 +234,33 @@ public String getId() {
         if (options == null) {
             options = new CosmosItemRequestOptions();
         }
+
+        if (!database.getClient().getTracerProvider().isEnabled()) {
+            return createItemInternal(item, options);
+        }
+
+        final CosmosItemRequestOptions requestOptions = options;
+        return withContext(context -> createItemInternal(item, requestOptions, context)).subscriberContext(TracerProvider.callDepthAttributeFunc);
+    }
+
+    private <T> Mono<CosmosAsyncItemResponse<T>> createItemInternal(T item, CosmosItemRequestOptions options, Context context) {
+        String spanName = "createItem." + this.getId();
```

Comment:
id is the id of the container. so it means in the scope of this container `"createItem." + this.getId()` can be pre-computed for the container, rather than re-computing it each time.

Could you pre-compute this in the container level.